graphql-bal-service/src/main/java/io/ballerina/graphql/schema/utils/Utils.java

```diff
@@ -0,0 +1,340 @@
+import io.ballerina.compiler.syntax.tree.MappingConstructorExpressionNode;
+import io.ballerina.compiler.syntax.tree.MappingFieldNode;
+import io.ballerina.compiler.syntax.tree.MetadataNode;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.NodeList;
+import io.ballerina.compiler.syntax.tree.ObjectConstructorExpressionNode;
+import io.ballerina.compiler.syntax.tree.QualifiedNameReferenceNode;
+import io.ballerina.compiler.syntax.tree.SeparatedNodeList;
+import io.ballerina.compiler.syntax.tree.ServiceDeclarationNode;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.graphql.schema.diagnostic.DiagnosticMessages;
+import io.ballerina.graphql.schema.exception.SchemaGenerationException;
+import io.ballerina.stdlib.graphql.commons.types.Schema;
+import io.ballerina.tools.diagnostics.Location;
+import io.ballerina.tools.text.LinePosition;
+import io.ballerina.tools.text.LineRange;
+import io.ballerina.tools.text.TextRange;
+import org.apache.commons.io.FilenameUtils;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static io.ballerina.graphql.schema.Constants.EMPTY_STRING;
+import static io.ballerina.graphql.schema.Constants.GRAPHQL_EXTENSION;
+import static io.ballerina.graphql.schema.Constants.MESSAGE_CANNOT_READ_SCHEMA_STRING;
+import static io.ballerina.graphql.schema.Constants.MESSAGE_INVALID_SCHEMA_STRING;
+import static io.ballerina.graphql.schema.Constants.MESSAGE_MISSING_ANNOTATION;
+import static io.ballerina.graphql.schema.Constants.MESSAGE_MISSING_FIELD_SCHEMA_STRING;
+import static io.ballerina.graphql.schema.Constants.MESSAGE_MISSING_SERVICE_CONFIG;
+import static io.ballerina.graphql.schema.Constants.PERIOD;
+import static io.ballerina.graphql.schema.Constants.SCHEMA_PREFIX;
+import static io.ballerina.graphql.schema.Constants.SCHEMA_STRING_FIELD;
+import static io.ballerina.graphql.schema.Constants.SERVICE_CONFIG_IDENTIFIER;
+import static io.ballerina.graphql.schema.Constants.SLASH;
+import static io.ballerina.graphql.schema.Constants.UNDERSCORE;
+import static io.ballerina.stdlib.graphql.commons.utils.Utils.PACKAGE_NAME;
+import static io.ballerina.stdlib.graphql.commons.utils.Utils.hasGraphqlListener;
+import static io.ballerina.stdlib.graphql.commons.utils.Utils.removeEscapeCharacter;
+
+/**
+ * Utility class for Ballerina GraphQL SDL schema generation.
+ */
+public class Utils {
+
+    /**
+     * Check whether the given service declaration node is related to a GraphQL service.
+     */
+    public static boolean isGraphqlService(ServiceDeclarationNode node, SemanticModel semanticModel) {
```

Comment:
Shouldn't we use the `graphql-commons` method for this, instead of duplicating the code?